series/series_test.go

```diff
@@ -1165,3 +1177,295 @@ func TestSeries_IsNaN(t *testing.T) {
+	tests := []struct {
+		series   Series
+		expected float64
+	}{
+		{
+			Ints([]int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}),
+			5.5,
+		},
+		{
+			Floats([]float64{1.0, 2.0, 3.0}),
+			2.0,
+		},
+		{
+			Strings([]string{"A", "B", "C", "D"}),
+			math.NaN(),
+		},
+		{
+			Bools([]bool{true, true, false, true}),
+			0.75,
+		},
+		{
+			Floats([]float64{}),
+			math.NaN(),
+		},
+	}
+
+	for testnum, test := range tests {
+		received := test.series.Mean()
+		expected := test.expected
+		if !compareFloats(received, expected, 6) {
+			t.Errorf(
+				"Test:%v\nExpected:\n%v\nReceived:\n%v",
+				testnum, expected, received,
+			)
+		}
+	}
+}
+
+func TestSeries_Max(t *testing.T) {
+	tests := []struct {
+		series   Series
+		expected float64
+	}{
+		{
+			Ints([]int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}),
+			10,
+		},
+		{
+			Floats([]float64{1.0, 2.0, 3.0}),
+			3.0,
+		},
+		{
+			Strings([]string{"A", "B", "C", "D"}),
+			math.NaN(),
+		},
+		{
+			Bools([]bool{true, true, false, true}),
+			1.0,
+		},
+		{
+			Floats([]float64{}),
+			math.NaN(),
+		},
+	}
+
+	for testnum, test := range tests {
+		received := test.series.Max()
+		expected := test.expected
+		if !compareFloats(received, expected, 6) {
+			t.Errorf(
+				"Test:%v\nExpected:\n%v\nReceived:\n%v",
+				testnum, expected, received,
+			)
+		}
+	}
+}
+
+func TestSeries_Min(t *testing.T) {
+	tests := []struct {
+		series   Series
+		expected float64
+	}{
+		{
+			Ints([]int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}),
+			1.0,
+		},
+		{
+			Floats([]float64{1.0, 2.0, 3.0}),
+			1.0,
+		},
+		{
+			Strings([]string{"A", "B", "C", "D"}),
+			math.NaN(),
+		},
+		{
+			Bools([]bool{true, true, false, true}),
+			0.0,
+		},
+		{
+			Floats([]float64{}),
+			math.NaN(),
+		},
+	}
+
+	for testnum, test := range tests {
+		received := test.series.Min()
+		expected := test.expected
+		if !compareFloats(received, expected, 6) {
+			t.Errorf(
+				"Test:%v\nExpected:\n%v\nReceived:\n%v",
+				testnum, expected, received,
+			)
+		}
+	}
+}
+
+func TestSeries_MaxStr(t *testing.T) {
+	tests := []struct {
+		series   Series
+		expected string
+	}{
+		{
+			Ints([]int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}),
+			"",
+		},
+		{
+			Floats([]float64{1.0, 2.0, 3.0}),
+			"",
+		},
+		{
+			Strings([]string{"A", "B", "C", "D"}),
+			"D",
+		},
+		{
+			Strings([]string{"quick", "Brown", "fox", "Lazy", "dog"}),
+			"quick",
+		},
+		{
+			Bools([]bool{true, true, false, true}),
+			"",
+		},
+		{
+			Floats([]float64{}),
+			"",
+		},
+	}
+
+	for testnum, test := range tests {
+		received := test.series.MaxStr()
+		expected := test.expected
+		if received != expected {
+			t.Errorf(
+				"Test:%v\nExpected:\n%v\nReceived:\n%v",
+				testnum, expected, received,
+			)
+		}
+	}
+}
+
+func TestSeries_MinStr(t *testing.T) {
+	tests := []struct {
+		series   Series
+		expected string
+	}{
+		{
+			Ints([]int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10}),
+			"",
+		},
+		{
+			Floats([]float64{1.0, 2.0, 3.0}),
+			"",
+		},
+		{
```

Comment:
Ouch! That's embarrassing... sorry about that, I could swear I've run those tests before pushing the changes. My bad. Fixed now.